scripts/import-cp-contexts.sh

```diff
@@ -0,0 +1,246 @@
+    user=$(KUBECONFIG="$in_KUBECONFIG" kubectl --kubeconfig <(echo "$kubeconfig") config get-users | tail -n +2)
+    kubeconfig=$(
+        echo "$kubeconfig" \
+        | yq ".clusters[0].name = \"$cp_name-cluster\"" \
+        | yq ".users[0].name = \"$cp_name-user\"" \
+        | yq ".contexts[0].name = \"$cp_name\"" \
+        | yq ".contexts[0].context.cluster = \"$cp_name-cluster\"" \
+        | yq ".contexts[0].context.user = \"$cp_name-user\"" \
+        | yq ".current-context = \"$cp_name\""
+    )
+
+    # swap out 127.0.0.1 with an external ip address
+    if [[ "$ip_addr" != "" ]] ; then
+        echov "- Replacing server ip address \"127.0.0.1\" with \"$ip_addr\""
+        kubeconfig=$(echo "$kubeconfig" | sed "s@server: https://127\.0\.0\.1:@server: https://$ip_addr:@g")
+    fi
+
+    # return the kubecconfig of the control plane in plain text
+    echo "$kubeconfig"
+}
+
+
+while (( $# > 0 )); do
+    case "$1" in
+    (--names|-n)
+        if (( $# > 1 ));
+        then { arg_names="$2"; shift; }
+        else { echo "$0: missing list of KubeFlex Control Plane names" >&2; exit 1; }
+        fi;;
+    (--kubeconfig|-k)
+        if (( $# > 1 ));
+        then { arg_kubeconfig="$2"; shift; }
+        else { echo "$0: missing kubeconfig filename" >&2; exit 1; }
+        fi;;
+    (--context|-c)
+        if (( $# > 1 ));
+        then { arg_context="$2"; shift; }
+        else { echo "$0: missing context name" >&2; exit 1; }
+        fi;;
+    (--replace-local-ip-address|-r)
+        if (( $# > 1 ));
+        then { arg_ip_addr="$2"; shift; }
+        else { echo "$0: missing ip address" >&2; exit 1; }
+        fi;;
+    (--merge|-m)
+        arg_merge=true;;
+    (--output|-o)
+        if (( $# > 1 ));
+        then { arg_output="$2"; shift; }
+        else { echo "$0: missing output filename" >&2; exit 1; }
+        fi;;
+    (--silent|-s)
+        arg_verbose=false;;
+    (-X)
+        set -x;;
+    (-h|--help)
+        display_help
+        exit 0;;
+    (-*)
+        echo "$0: unknown flag" >&2
+        exit 1;;
+    (*)
+        echo "$0: unknown positional argument" >&2
+        exit 1;;
+    esac
+    shift
+done
+
+
+# Define the echov function based on verbosity
+if [ "$arg_verbose" == "true" ]; then
+    echov() { echo "$@" ; }
+else
+    echov() { :; }
+fi
+
+
+# Determine the list of kubeconfigs to search
+# based on https://kubernetes.io/docs/reference/KUBECONFIG="$in_KUBECONFIG" kubectl/generated/KUBECONFIG="$in_KUBECONFIG" kubectl_config/
+echov "Determining the list of kubeconfigs to search:"
+if [[ "$arg_kubeconfig" != "" ]] ; then
+    in_KUBECONFIG="$arg_kubeconfig"
+    if [[ "$arg_output" == "" ]] ; then
+        out_kubeconfig="$arg_kubeconfig"
+    fi
+elif [[ "$KUBECONFIG" != "" ]] ; then
+    in_KUBECONFIG="$KUBECONFIG"
+    if [[ "$arg_output" == "" ]] ; then
+        if [[ "$KUBECONFIG" == *":"* ]] ; then
+            out_kubeconfig="$HOME/.kube/config" # hard to decide
```

Comment:
Yeah, I would err out in this case, because the user's intent is not sufficiently obvious enough for any guess to be safe.